### DISCUSSION TITLE
Welcome to ant-design Discussions!

### DISCUSSION THREAD

**Discussion by chenshuai2144** [ant-design/ant-design#28265] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ycjcl868** [comment]
What is the difference between issues and discussions?😁

**Reply by yaxiaoliu** [reply]
LGTM

**Comment by xrkffgg** [comment]
Feels good.

**Comment by vagusX** [comment]
火前留名🔥

**Comment by howel52** [comment]
[make opensource great again](https://gw.alipayobjects.com/mdn/rms_38ea80/afts/img/A*3IWGQLAmPZwAAAAAAAAAAAAAARQnAQ)

**Comment by hengkx** [comment]
☺️

**Comment by titanics2008** [comment]
mark。New？

**Comment by mane115** [comment]
🚀🚀

**Comment by huntdream** [comment]
😄 look look

**Comment by kimmy-wang** [comment]
look look

**Comment by yingpengsha** [comment]
github tieba🤨

**Comment by mxppxm** [comment]


**Reply by xrkffgg** [reply]
正常的啊

**Comment by janyin** [comment]
讨论区这个功能很强

**Comment by travonf** [comment]
打开讨论区感觉跟issue区没啥区别啊，看标题大部分都是使用问题

**Reply by chenshuai2144** [reply]
这个可以单独回复的，还是很有用

**Comment by lkhchen** [comment]
> ## 👋 Welcome!
> We’re using Discussions as a place to connect with other members of our community. We hope that you:
> 
> * Ask questions you’re wondering about.
> * Share ideas.
> * Engage with other community members.
> * Welcome others and are open-minded. Remember that this is a community we
>   build together 💪.
> 
> To get started, comment below with an introduction of yourself and tell us about what you do with this community.

有问题想问希望这里能有答案

**Comment by prabhat-gc** [comment]
May i know why antd version 4 didn't works from shared project. it works fine if we do dierct import from antd v4. And antd v3 is also working fine with shared component. I always getting `hook error` for each antd v4 component. 
you can also try by antd v4 & v3 component in `sahted-libs` of https://github.com/doronoded/shared-lib-example

**Comment by rudyzou** [comment]
请问5什么时候发布？

**Comment by kiner-tang** [comment]
`ant-design`下个版本的规划有定么？在哪里可以看呀

**Comment by branden-jones** [comment]
Hey Everyone,
My name is Branden Jones and im working with a non-profit to build them an application that allows admin to match mentee and mentors together for organization. I ran into an issue I need help with...

~ our database has a lot of info, so at various stages (in order to reduce API call) we are storing data in state using redux. I have access to all the data I need this way and the page renders quickly...

I want to utilize Ant Designs <Tag /> component to show when users have 0 matches... 

Below is the Table where I am taking the users ( an Array of Object ) and displaying to the webpage, but MY QUESTION:
is there a way to give the <Tag /> component to a specific property with each object...

<Table columns={columns} dataSource={allUsers} />

Maybe I can store the <Tag /> component with Redux??

**Comment by huli66** [comment]
穿梭框为什么采用三个属性：dataSource, targetKeys, selectedKeys，来代表穿梭框左右两边的内容，然后再在内部把 selectedKeys 拆分为两个 state，而不是直接传入targetSelectKeys和sourceSelectedKeys属性，这样做是出于什么考虑呢
```js
const [sourceSelectedKeys, setSourceSelectedKeys] = useState<string[]>(() =>
    selectedKeys.filter((key) => !targetKeys.includes(key))
  );

  const [targetSelectedKeys, setTargetSelectedKeys] = useState<string[]>(() =>
    selectedKeys.filter((key) => targetKeys.includes(key))
  );

  useEffect(() => {
    if (props.selectedKeys) {
      setSourceSelectedKeys(() =>
        selectedKeys.filter((key) => !targetKeys.includes(key))
      );
      setTargetSelectedKeys(() =>
        selectedKeys.filter((key) => targetKeys.includes(key))
      );
    }
  }, [props.selectedKeys, props.targetKeys]);
```

**Comment by ghost** [comment]
Hello!
